RFID.py

```diff
@@ -25,20 +25,14 @@ def readProximityState():
 
     def listenForRFID():
 
-        RFID_code = ""
-
-        while True:
+        RFID_code = []
+        byte = ''
 
+        while byte != '\r':
             byte = self.readByte()
+            RFID_code.append(byte)
 
-            if byte == '\r':
-
-                RFID_code = RFID_code[2:len(RFID_code) - 1] #TODO parse RFID code properly
-                self.serial_interface.reset_input_buffer()
-                sleep(1)
-                return RFID_code 
-            else:
-
-                RFID_code += byte
-
-
+        self.serial_interface.reset_input_buffer()
+        sleep(1)
+        
+        return ''.join(RFID_code[2:])
```

Comment:
Not that it really matters here, but this is a big performance improvement since strings are immutable in Python